admin/tool/policy/lib.php

```diff
@@ -68,3 +69,21 @@ function tool_policy_before_standard_html_head() {
 
     return $message;
 }
+
+/**
+ * Hooks redirections to digital minor validation and policy acceptance pages before sign up.
+ */
+function tool_policy_pre_signup_requests() {
+
+    if (!isset($SESSION->minor)) {  // Digital minor check hasn't been done.
```

Comment:
Plugins should not pollute the SESSION freely. If you need something there, put it into your plugin's namespace to avoid potential collisions, e.g. `$SESSION->tool_policy->minor` or at least `$SESSION->tool_policy_minor`.

Reply:
Also, we should invalidate the flag in the session after a while such as 30 minutes. It effectively prevents any other coming to the same browser to register on the site.